src/commands/trialApp/showTutorial.ts

```diff
@@ -15,6 +15,9 @@ export async function showTutorial(context: IActionContext): Promise<void> {
 
     if (trialAppNode) {
         const extensionId: string = 'redhat.vscode-didact';
+        if (!extensions.getExtension(extensionId)) {
```

Comment:
Since we're checking for getExtension here now, I think we don't actually need the check in `installExtension`.  I think `installExtension` should only be called if we need to install anyway.  The getExtension check inside the while loop should stay, but the one before `executeCommand(extension.open)` can go.